src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -664,12 +664,11 @@ public function deleteAction(Request $request)
      * )
      * @DemoRestricted(redirectRoute="admin_categories_index")
      *
-     * @param CategoryFilters $filters
-     *
      * @return Response
      */
-    public function exportAction(CategoryFilters $filters)
+    public function exportAction()
     {
+        $filters = new CategoryFilters(['limit' => null] + CategoryFilters::getDefaults());
```

Comment:
By doing that you ignore all previously set filters in the grid, so the export action will simply return all the categories all the time (regardless of your current filtering) I'd recommend rebuild an object filter based on the provided values:
```php
$filters = new CategoryFilters($filters->all() + ['limit' => null], $filters->getFilterId());
```